queries.sql

```diff
@@ -16,4 +16,61 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
 
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, MAX(escape_attempts) AS max_escape_attempts
+FROM animals
+GROUP BY neutered;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
+
+-- Transaction 1: Update species to 'unspecified' and rollback
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+-- Verify the change
+SELECT * FROM animals;
+-- Rollback
+ROLLBACK;
+
+-- Transaction 2: Update species based on name and commit
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+-- Verify the changes
+SELECT * FROM animals;
+-- Commit the transaction
+COMMIT;
+
+-- Transaction 3: Delete all records and rollback
+BEGIN;
+DELETE FROM animals;
+-- Verify if all records in the 'animals' table still exist (empty result set)
+SELECT * FROM animals;
+-- Rollback
+ROLLBACK;
+
```

Comment:
- [ ] According to the requirements you should check if the records in the animals table still exist.